timeseries/src/autogluon/timeseries/models/ensemble/array_based/abstract.py

```diff
@@ -0,0 +1,236 @@
+from abc import ABC
+from typing import Any, Optional, Sequence, Type, Union
+
+import numpy as np
+
+from autogluon.timeseries.dataset import TimeSeriesDataFrame
+from autogluon.timeseries.metrics.abstract import TimeSeriesScorer
+from autogluon.timeseries.utils.features import CovariateMetadata
+
+from ..abstract import AbstractTimeSeriesEnsembleModel
+from .regressor import EnsembleRegressor
+
+
+class ArrayBasedTimeSeriesEnsembleModel(AbstractTimeSeriesEnsembleModel, ABC):
+    """Abstract base class for time series ensemble models which operate on arrays of base model
+    predictions for training and inference.
+
+    Other Parameters
+    ----------------
+    isotonization: str, default = "sort"
+        The isotonization method to use (i.e. the algorithm to prevent quantile non-crossing).
+        Currently only "sort" is supported.
+    detect_and_ignore_failures: bool, default = True
+        Whether to detect and ignore "failed models", defined as models which have a loss that is larger
+        than 10x the median loss of all the models. This can be very important for the regression-based
+        ensembles, as moving the weight from such a "failed model" to zero can require a long training
+        time.
+    """
+
+    _regressor_type: Type[EnsembleRegressor]
+
+    def __init__(
+        self,
+        path: Optional[str] = None,
+        name: Optional[str] = None,
+        hyperparameters: Optional[dict[str, Any]] = None,
+        freq: Optional[str] = None,
+        prediction_length: int = 1,
+        covariate_metadata: Optional[CovariateMetadata] = None,
+        target: str = "target",
+        quantile_levels: Sequence[float] = (0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9),
+        eval_metric: Union[str, TimeSeriesScorer, None] = None,
+    ):
+        super().__init__(
+            path=path,
+            name=name,
+            hyperparameters=hyperparameters,
+            freq=freq,
+            prediction_length=prediction_length,
+            covariate_metadata=covariate_metadata,
+            target=target,
+            quantile_levels=quantile_levels,
+            eval_metric=eval_metric,
+        )
+        self.ensemble_regressor: Optional[EnsembleRegressor] = None
+        self._model_names: list[str] = []
+
+    def _get_default_hyperparameters(self) -> dict[str, Any]:
+        return {
+            "isotonization": "sort",
+            "detect_and_ignore_failures": True,
+        }
+
+    @staticmethod
+    def to_array(df: TimeSeriesDataFrame) -> np.ndarray:
+        """Given a TimeSeriesDataFrame object, or a list or dict of such objects, return
+        a single array composing the values contained in the data frames.
+
+        Parameters
+        ----------
+        df
+            TimeSeriesDataFrame to convert to an array.
+
+        Returns
+        -------
+        array
+            of shape (num_items, num_timesteps, num_quantiles).
```

Comment:
How about we explicitly replace `num_timesteps` with `prediction_length` here? As I understand, there is an implicit assumption that `df` is a TSDF of predictions, where each item has `prediction_length` values (otherwise reshaping to an array won't work).

Another option is to rename the argument `df` -> `predictions`.